libglusterfs/src/mem-pool.c

```diff
@@ -137,23 +130,14 @@ gf_mem_update_acct_info(struct mem_acct *mem_acct, struct mem_header *header,
 
     if (mem_acct != NULL) {
         rec = &mem_acct->rec[header->type];
+        rec->max_size = max(rec->max_size, rec->size);
+#ifdef DEBUG
         LOCK(&rec->lock);
         {
-            rec->size += size - header->size;
-            rec->total_allocs++;
-            rec->max_size = max(rec->max_size, rec->size);
-
-#ifdef DEBUG
-            /* The old 'header' already was present in 'obj_list', but
-             * realloc() could have changed its address. We need to remove
-             * the old item from the list and add the new one. This can be
-             * done this way because list_move() doesn't use the pointers
-             * to the old location (which are not valid anymore) already
-             * present in the list, it simply overwrites them. */
```

Comment:
I would keep this comment. Otherwise it's hard to understand the meaning of this list_move().

Reply:
done